Abyss.Web/Controllers/ServerController.cs

```diff
@@ -25,15 +31,61 @@ public async Task<List<Server>> GetServers()
 
     [Route("start/{serverId}")]
     [HttpPost]
-    public async Task Start(int serverId)
+    public IActionResult Start(int serverId, [Required][FromQuery] string connectionId)
     {
-        await _serverManager.Start(serverId);
+        _backgroundTaskQueue.Queue(async (serviceProvider, cancellationToken) =>
+        {
+            var serverManagerHub = serviceProvider.GetRequiredService<IHubContext<ServerManagerHub>>();
+            var scopedServerManager = serviceProvider.GetRequiredService<IServerManager>();
+            var client = serverManagerHub.Clients.Clients(connectionId);
+            string err = null;
+            try
+            {
+                await scopedServerManager.Start(serverId, async (logItem) =>
+                {
+                    await client.SendAsync("log", logItem.Message);
+                }, async (status) =>
+                {
+                    await client.SendAsync("status", status);
+                });
+            }
+            catch (Exception e)
+            {
+                err = e.Message;
+            }
+            await client.SendAsync("complete", err);
```

Comment:
You are potentailly exposing the inner workings of the system to the user here.

Reply:
For my use case I'm not bothered about this but in a proper production system agree